tests/regression.rs

```diff
@@ -0,0 +1,151 @@
+//! High-level regression testing which will build the example book and *try*
+//! to make sure key elements don't get accidentally broken.
+//!
+//! # Warning
+//!
+//! These tests will need to be updated every time the example book changes.
+//! Hopefully Travis will let you know when that happens.
+
+
+#[macro_use]
+extern crate pretty_assertions;
+extern crate select;
+extern crate walkdir;
+
+mod dummy;
+
+use std::path::Path;
+use walkdir::{DirEntry, WalkDir, WalkDirIterator};
+use select::document::Document;
+use select::predicate::{Class, Name, Predicate};
+
+
+const BOOK_ROOT: &'static str = concat!(env!("CARGO_MANIFEST_DIR"), "/book-example");
```

Comment:
I would suggest to using something more stable than the book-example. It is our de facto documentation which will naturally grow and having it rigorously tested against static tests might make a little painful to edit.
How about using something a little more synthetic and static with which we could test several ugly edge-cases and for instance check if:
- playpens with given info-strings results in required css classes
- if js libs are imported if given option in book.toml is given.
- ???

Actually, we might consider using some non rust solutions for web regression testing (I'm no webdev but I've seen some really impressive magic done with selenium)

Reply:
Yeah, my first thought was to use Python and BeautifulSoup because it makes it loads easier to navigate HTML documents and find the things you want, I was a little hesitant to add more complication and a second language to the testing process though. I'll have a look at what other projects do to verify rendered HTML/CSS and check for regressions.

In terms of making things more static, would it be a good idea to rewrite this based on the "dummy book" instead? It'd mean the integration tests need to be updated whenever you change it so you can test for regressions, but that should give a nicer experience than tests breaking whenever we update documentation.

Reply:
There are libraries similar to BeautifulSoup in Rust

Reply:
> In terms of making things more static, would it be a good idea to rewrite this based on the "dummy book" instead?

Yeah. Im not even sure we need to create it dynamically. Why not have the dummy book as a directory structure fixture similar to book-example which we could easily expand 

> There are libraries similar to BeautifulSoup in Rust

These tests already use the [`select`](https://crates.io/crates/select) (nice wrapper over `html5ever`) crate which is as close to bs4 as we can get now. I was thinking more in terms of actual usability tests with headless [selenium](http://www.seleniumhq.org) but might be too ambitious for now.

Reply:
> Why not have the dummy book as a directory structure fixture similar to book-example which we could easily expand

I like this idea. Originally I was dynamically generating the dummy book in a temporary directory so each test is completely independent from the others, but there's no reason why we can't just do a `cp -r` into a temp directory.